### ISSUE TITLE
Failing to publish results in JIRA Xray 

### ISSUE THREAD

**Issue by ramran-r** [fundakol/pytest-jira-xray#25]
Getting below error....

ERROR    pytest_xray.xray_publisher:xray_publisher.py:91 HTTPError: Could not post to JIRA service at https://<JIRA URL>/rest/raven/2.0/import/execution. Response status code: 400
requests.exceptions.HTTPError: 400 Client Error:  for url: https://<JIRA URL>/rest/raven/2.0/import/execution
ERROR    pytest_xray.xray_publisher:xray_publisher.py:93 Error message from server: issuetype: issuetype

Defined below syntax in 
       @pytest.mark.xray('SECSOLTEST-291')

All environment variables are defined except client ID & Client secret key. Are these mandate?

**Comment by ramran-r** [comment]
With passing execution ID , it works well --- pytest --jira-xray  --execution SECSOL-360  
However using "pytest --jira-xray" alone, it fails to create new test execution and publish the result . Not sure how it can identify the project where the 'test execution' issue type should be created. 
Getting below error if you dont pass 'execution ID' ...Can you help to resolve this ?
Could not publish results to Jira XRAY!
HTTPError: Could not post to JIRA service at https://jira01.engit.synamedia.com/rest/raven/2.0/import/execution. Response status code: 400

**Comment by fundakol** [comment]
Can you run your tests with command: 
```
pytest --jira-xray --xraypath=xray.json
```
a then upload xray.json manually from Jira UI. It may give you more detailed description about the error. 
Error `issuetype` tells me nothing.

**Comment by grssmnn** [comment]
@ramran-r I got the same issue if `XRAY_EXECUTION_FIX_VERSION` is set.

Full error message from API is saying:
```{'error': "Error assembling issue data: Version name '1.0' is not valid"}```

The version must be also specified in Jira itself, otherwise the request will fail.
Setting the version is not mandatory, therefore you can also leave it out.

**Comment by ramran-r** [comment]
@fundakol - Does this command ( pytest --jira-xray )  create a new issue type as 'Test Execution'? I had success run only if i pass the execution ID created separately . 
@grssmnn - I have ignored the variable  XRAY_EXECUTION_FIX_VERSION as it is defined as 'none' in my JIRA configuration.

**Comment by lozik4** [comment]
Could not publish results to Jira XRAY!
HTTPError: Could not post to JIRA service at https://show4me.atlassian.net/rest/raven/2.0/import/execution. Response status code: 404

I Use all parametrs resalt:

**Comment by mklankireddy** [comment]
@fundakol  i have used pytest --jira-xray --xraypath=xray.json,
after running the tests my test status is set to either PASS/FAIL.
JIRA on the other hand is expecting PASSED/FAILED, and when i try to upload test results manually my test status wont change.

**Comment by fundakol** [comment]
@mklankireddy 
There are two endpoints for xray, Cloud and Server+DC https://docs.getxray.app/display/XRAY/REST+API
They use different statuses for results (PASS vs PASSED), so if you want to use the xray plugin for could server you need to pass additional argument ``--cloud``, then it will generated proper report for the cloud.

**Comment by mklankireddy** [comment]
awesome !!, --jira-xray --cloud --xraypath=xray.json worked. Thanks @fundakol

**Comment by mklankireddy** [comment]
i was having issue with importing test results to XRAY cloud, 

### environment variables i had setup:
- `XRAY_API_BASE_URL=https://xray.cloud.getxray.app/` - set my base URL according to [XRAY documentation](https://docs.getxray.app/display/XRAYCLOUD/REST+API)
- `XRAY_API_KEY=yourapikey`

### added the following for my test:
`@pytest.mark.xray('ABC-456')` added my test case id 

using `pytest --jira-xray --api-key-auth --execution ABC-123` to run my test and upload results to XRAY cloud caused an error because i missed `--cloud` option. since i am trying to import results to XRAY cloud and deal with XRAY cloud in general in my case i had to use `--cloud` for both importing results to XRAY cloud and generate json file with results file compatible with XRAY cloud.

### Finally commands that worked for me
- command to create json file `pytest --jira-xray --cloud --xraypath=xray.json`
- command to import test results to XRAY cloud `pytest --jira-xray --api-key-auth --cloud --execution ABC-123`

**Comment by pfunk1978** [comment]
@fundakol Yes. It was the "Test Plan" field, somehow got removed from the Test Execution create screen. I'm calling shenanigans on a workmate.
